pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -230,8 +230,8 @@ func (a *Actuator) CreateMachine(cluster *clusterv1.Cluster, machine *clusterv1.
 func (a *Actuator) Delete(context context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	glog.Info("deleting machine")
 	if err := a.DeleteMachine(cluster, machine); err != nil {
-		glog.Errorf("error deleting machine: %v", err)
-		return err
+		glog.Errorf("error deleting machine: %v, returning an error to requeue", err)
+		return &clustererror.RequeueAfterError{RequeueAfter: requeueAfterSeconds * time.Second}
```

Comment:
In all failure cases Delete() means retry?

Reply:
This should probably only happen when the instance state is not:
	// InstanceStateNameTerminated is a InstanceStateName enum value
	InstanceStateNameTerminated = "terminated"

Reply:
It will always retry by returning an error as it will be requeued, that's normal/expected controller behaviour. The intention of requeue error is to support long running operations by requeing after a known period of time in specific situations e.g `InstanceStateNameShuttingDown = "shutting-down"`

Reply:
If the instance simply doesn't exist then there's no point retrying. It seems to me we don't handle this state, and will needlessly retry forever.

Reply:
The assumption is that at some point (right away, in the first/second/third iteration) you will be able to:
- decode machine provider config to get instance ID
- build aws client to query for running instances
- get running instances owned by a machine object
- terminate instance (if there is any)

So eventually, every running instances to be deleted gets deleted. If a machine gets re-queued too many times, there is something wrong with the infrastructure and additional actions needs to be taken. 

Additionally, if an instance is in `shutdown` or `terminated` state, it will not get listed in the list of running instances. Unless we want to react on the fact an instance is in `shutdown` state for a long time. 

@enxebre @frobware so is there anything else I don't see that needs to be captured by the on-delete-error requeue logic?